gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/DryRunConfigurationCacheEnabledTask.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.configcache.incremental;
+
+import com.google.common.base.Splitter;
+import com.palantir.gradle.utils.circleciartifacts.ArtifactLocation;
+import com.palantir.gradle.utils.circleciartifacts.CircleCiArtifacts;
+import com.palantir.gradle.utils.environmentvariables.EnvironmentVariables;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.TaskAction;
+
+public abstract class DryRunConfigurationCacheEnabledTask extends AbstractDryRunTask {
+
+    @Nested
+    protected abstract EnvironmentVariables getEnvironmentVariables();
+
+    @Nested
+    protected abstract CircleCiArtifacts getCircleCiArtifacts();
+
+    public DryRunConfigurationCacheEnabledTask() {
+        getMarkerOutputFile()
+                .set(getTemporaryDir()
+                        .toPath()
+                        .resolve("dryRunConfigurationCacheEnabled.marker")
+                        .toFile());
+    }
+
+    @TaskAction
+    public final void check() throws IOException {
+        DryRunResult result =
+                dryRun(List.of("--configuration-cache", "-Pconfiguration-cache-compatible-for-all-tasks"));
```

Comment:
> -Pconfiguration-cache-compatible-for-all-tasks

Let's explain why we want this? It's not immediately obvious from reading the code